pkg/cmd/grafana-cli/services/api_client.go

```diff
@@ -72,19 +84,27 @@ func (client *GrafanaComClient) DownloadFile(pluginName, filePath, url string, c
 		}
 	}()
 
-	// TODO: this would be better if it was streamed file by file instead of buffered.
 	// Using no timeout here as some plugins can be bigger and smaller timeout would prevent to download a plugin on
 	// slow network. As this is CLI operation hanging is not a big of an issue as user can just abort.
-	body, err := sendRequest(HttpClientNoTimeout, url)
-
+	bodyReader, err := executeRequestReader(HttpClientNoTimeout, url)
 	if err != nil {
-		return nil, errutil.Wrap("Failed to send request", err)
+		return errutil.Wrap("Failed to send request", err)
 	}
+	defer bodyReader.Close()
 
-	if len(checksum) > 0 && checksum != fmt.Sprintf("%x", md5.Sum(body)) {
-		return nil, xerrors.New("Expected MD5 checksum does not match the downloaded archive. Please contact security@grafana.com.")
+	w := bufio.NewWriter(tmpFile)
+	h := md5.New()
+	if _, err = io.Copy(w, io.TeeReader(bodyReader, h)); err != nil {
+		return errutil.Wrap("Failed to compute MD5 checksum", err)
 	}
-	return body, nil
+	w.Flush()
+	if err := tmpFile.Close(); err != nil {
```

Comment:
I would think it's better to leave the closing of the tempfile to the function that opened it (via `defer`).

Reply:
Done